pages/admin/lessons/dojo/modules.tsx

```diff
@@ -0,0 +1,126 @@
+import { gql, useQuery } from '@apollo/client'
+import { GetAppProps, withGetApp } from '../../../../graphql'
+import { get, sortBy } from 'lodash'
+import React, { useEffect, useMemo, useState } from 'react'
+import AdminLessonNav from '../../../../components/admin/lessons/AdminLessonNav'
+import AdminLessonSideNav from '../../../../components/admin/lessons/AdminLessonSideNav'
+import AdminModuleInputs, {
+  Props
+} from '../../../../components/admin/lessons/AdminModuleInputs'
+import Breadcrumbs from '../../../../components/Breadcrumbs'
+import styles from '../../../../scss/modules.module.scss'
+import { AdminLayout } from '../../../../components/admin/AdminLayout'
+
+const MODULES = gql`
+  query {
+    modules {
+      id
+      name
+      content
+      lesson {
+        id
+      }
+    }
+  }
+`
+
+type Module = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+}
+type Modules = Module[]
+
+type ModulesTabProps = {
+  filteredModules: Modules
+  lessonId: number
+  refetch: Props['refetch']
+}
+
+const ModulesTab = ({
+  filteredModules,
+  lessonId,
+  refetch
+}: ModulesTabProps) => {
+  const [selectedIndex, setSelectedIndex] = useState(-1)
+  const onAddItem = () => setSelectedIndex(-1)
+  const onSelect = (item: Module) => setSelectedIndex(item.id)
+
+  useEffect(() => setSelectedIndex(filteredModules[0]?.id), [lessonId])
+
+  return (
+    <div className={styles.container__modulesPanel}>
+      <AdminLessonSideNav
+        title="Modules"
+        items={filteredModules}
+        onAddItem={onAddItem}
+        onSelect={onSelect}
+        selectedIndex={selectedIndex}
+      />
+      <div className={styles.container__modulesPanel__inputs}>
+        <AdminModuleInputs
+          lessonId={lessonId}
+          refetch={refetch}
+          module={filteredModules.find(module => module.id === selectedIndex)}
+        />
+      </div>
+    </div>
+  )
+}
+
+const Lessons = ({ data }: GetAppProps) => {
+  const { lessons } = data
+  const { data: modulesData, refetch } = useQuery<{ modules: Modules }>(MODULES)
```

Comment:
Is it possible to set a default value for `modulesData` so that we don't have to use optional chaining or lodash `get` ?

Reply:
Same with the lessons variable, can you look into trying to set a default value for that as well

Reply:
nevermind about lessons variable, it looks like we only extract a property from it one time in this component

Reply:
Hmm actually i'm not sure if its appropriate to set a default value for `modulesData` either since we only grab `modules.name` from it. I'll resolve this thread and leave it up for you to decide